sycl/include/CL/sycl/group_algorithm.hpp

```diff
@@ -161,13 +221,16 @@ reduce_over_group(Group g, T x, BinaryOperation binary_op) {
   return result;
 }
 
+//   four argument variant of reduce_over_group specialized twice
+//       (scalar arithmetic || complex), and vector_arithmetic
 template <typename Group, typename V, typename T, class BinaryOperation>
-detail::enable_if_t<(is_group_v<std::decay_t<Group>> &&
-                     detail::is_scalar_arithmetic<V>::value &&
-                     detail::is_scalar_arithmetic<T>::value &&
-                     detail::is_native_op<V, BinaryOperation>::value &&
-                     detail::is_native_op<T, BinaryOperation>::value),
-                    T>
+detail::enable_if_t<
+    (is_group_v<std::decay_t<Group>> &&
+     (detail::is_scalar_arithmetic<V>::value || detail::is_complex<V>::value) &&
+     (detail::is_scalar_arithmetic<T>::value || detail::is_complex<T>::value) &&
+     detail::is_native_op<V, BinaryOperation>::value &&
+     detail::is_native_op<T, BinaryOperation>::value),
+    T>
```

Comment:
Should the test here check that `complex` is only used with `plus`, similar to how you coded the test for `joint_reduce`?  I have the same question about the other specializations and other algorithms.

Reply:
Correct, they should. I was leaning on the innermost specializations, since these functions all just call each other, but that's not future proof.  Updated all these "outer" specializations.